### PR TITLE
Fix write_audio_16bit write_audio_24bit write_audio_32bit and write_audio_float function heap buffer overflow

### PR DESCRIPTION
**Pull request by cometails**
Fix write_audio_16bit write_audio_24bit write_audio_32bit and write_audio_float function heap buffer overflow

### CODE REVIEW

**Comment by fabiangreffrath** [comment]
@eustas What do you think?

**Comment by eustas** [comment]
Looks good. Will be more readable it we also make an increment to be `6`; also it should be either `i + 6 <= samples` or `i + 5 < samples`

**Comment by fabiangreffrath** [comment]
> also it should be either `i + 6 <= samples` or `i + 5 < samples`

Yes, that's what I thought. Thanks!

File: frontend/audio.c [hunk 1]
```diff
@@ -323,7 +323,7 @@ static size_t write_audio_16bit(audio_file *aufile, void *sample_buffer,
 
     if (aufile->channels == 6 && aufile->channelMask)
     {
-        for (i = 0; i < samples; i += aufile->channels)
+        for (i = 0; i+6 < samples; i += aufile->channels)
```
Comment (fabiangreffrath) [inline frontend/audio.c:326]: it should be either `i + 6 <= samples` or `i + 5 < samples`

**Review by fabiangreffrath — CHANGES_REQUESTED** [review]

File: frontend/audio.c (hunk 1 above)
Comment (cometails) [reply frontend/audio.c:326]: Ok, I will modify it according to the suggestions. Thank you very much！

**Pull request merged**